src/ray/gcs/gcs_server/object_info_handler_impl.cc

```diff
@@ -56,8 +57,16 @@ void DefaultObjectInfoHandler::HandleAddObjectLocation(
   RAY_LOG(DEBUG) << "Adding object location, job id = " << object_id.TaskId().JobId()
                  << ", object id = " << object_id << ", node id = " << node_id;
 
-  auto on_done = [object_id, node_id, reply, send_reply_callback](Status status) {
-    if (!status.ok()) {
+  auto on_done = [this, object_id, node_id, reply,
```

Comment:
Now we can just save the locations in `ObjectLocator`, instead of saving it in redis. 
And maybe we should rename `ObjectLocator` to `GcsObjectManager` to align with other GCS modules. Then we can remove this `DefaultObjectInfoHandler`, and let `GcsObjectManager` handle the RPC directly.
(This might require some changes. It's fine if you prefer doing these changes in next PR.)

Reply:
This is a good idea, i will do these changes in next PR.